C++/source/dynamic_programming/longest_increasing_subsequence.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+   Longest Increasing Subsequence
+   ------------------------------
+   Given an unsorted array of integers, return the longest increasing subsequence
+
+   Time complexity
+   --------------
+   O(N), where N is size of array
```

Comment:
I think it's *O*(N<sup>2</sup>) since for every index you call `find_previous_smaller()`, which takes linear time. So for `N` number of indices, it would take *O*(N<sup>2</sup>) time.